template/androidApp/src/main/java/io/bloco/template/features/details/DetailsViewModel.kt

```diff
@@ -18,21 +18,21 @@ class DetailsViewModel @AssistedInject constructor(
 ) : ViewModel() {
 
     private val _bookDetailsUpdateState =
-        MutableStateFlow<APIRequestState>(APIRequestState.LoadingFromAPI)
+        MutableStateFlow<DetailsScreenUiState>(DetailsScreenUiState.LoadingFromAPI)
     val bookDetailsUpdateState = _bookDetailsUpdateState.asStateFlow()
 
     init {
         viewModelScope.launch {
             getBook(bookId)
-                .onSuccess { _bookDetailsUpdateState.value = APIRequestState.Success(it) }
-                .onFailure { _bookDetailsUpdateState.value = APIRequestState.ErrorFromAPI }
+                .onSuccess { _bookDetailsUpdateState.value = DetailsScreenUiState.Success(it) }
+                .onFailure { _bookDetailsUpdateState.value = DetailsScreenUiState.ErrorFromAPI }
         }
     }
 
-    sealed class APIRequestState {
-        object LoadingFromAPI : APIRequestState()
-        data class Success(val book: BookDetails) : APIRequestState()
-        object ErrorFromAPI : APIRequestState()
+    sealed interface DetailsScreenUiState {
```

Comment:
Now we're not using sealed classes, only sealed interfaces?

Reply:
It was what was done [here](https://youtu.be/qX6zmKY4KP0?t=435), so yeh.. 😃